app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -331,10 +331,11 @@ def save_switches_inventory(host, hashes)
     host.switches.delete(deletes)
   end
 
-  def remove_obsolete_switches
+  def remove_obsolete_switches(old_switch_ids)
     # delete from switches as s where s.shared is NULL and s.id not in (select switch_id from host_switches)
     # delete from switches as s where s.shared = 't' and s.id not in (select switch_id from host_switches)
-    Switch.where.not(:id => HostSwitch.all.collect(&:switch).uniq).destroy_all
+    obsolete_switch_ids = old_switch_ids - HostSwitch.all.collect(&:switch_id)
```

Comment:
Since our old_switch_ids is scoped down to just this ems (as it should be) we should use `- ems.switches.reload.pluck(:id)`

Reply:
@agrare, shoud we consider obsolete only switches that not have relationship with hosts?

Reply:
Because we use `obsolete_switches - ems.switches.reload.pluck(:id)` we are only considering the difference between lists

Reply:
> Because if we use `obsolete_switches - ems.switches.reload.pluck(:id)` we are only considering the difference between lists

Yeah that's the idea, we basically want to replicate the behavior of using the association to do the deletes.

Reply:
oh, I got it. I will change this right now.